pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1598,4 +1598,131 @@ public void testNoParametersNPE() throws SQLException {
       // ignore
     }
   }
+
+  @Test
+  public void testCountForMutableStatementWithSet() throws SQLException {
+    //INSERT
+    PreparedStatement pstmt = con.prepareStatement("SET search_path = 'public';"
+        + "INSERT INTO inttable VALUES (?)");
+    pstmt.setInt(1, 1);
+    int count = pstmt.executeUpdate();
+    pstmt.close();
+    assertEquals("Expected update count to be 1", 1, count);
+    //With two SET statements
+    pstmt = con.prepareStatement("SET search_path = 'public';SET search_path = 'public';"
+        + "INSERT INTO inttable VALUES (?)");
+    pstmt.setInt(1, 1);
+    count = pstmt.executeUpdate();
+    pstmt.close();
+    assertEquals("Expected update count to be 1", 1, count);
```

Comment:
Could you please split the test into several methods?

Could you use try-with-resources for closing the resources?

Reply:
I had more tests and hope granularity is ok